src/EGraphs/egraph.jl

```diff
@@ -508,7 +508,7 @@ function rebuild!(g::EGraph)
   n_unions = process_unions!(g)
   trimmed_nodes = rebuild_classes!(g)
   # @assert check_memo(g)
-  # @assert check_analysis(g)
+  @assert check_analysis(g)
```

Comment:
Let's add some kwargs `should_check_analysis::Bool=false` (same for memo) and add them to `SaturationParams` so that they are passed down to rebuild. It should be configurable. If you need checking, you can then enable it on demand